components/builder-api/src/server/resources/settings.rs

```diff
@@ -168,6 +173,53 @@ fn update_origin_package_settings(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn delete_origin_package_settings(req: HttpRequest, path: Path<(String, String)>) -> HttpResponse {
+    let (origin, pkg) = path.into_inner();
+
+    let account_id =
+        match authorize_session(&req, Some(&origin), Some(OriginMemberRole::Maintainer)) {
+            Ok(session) => session.get_id(),
+            Err(err) => return err.into(),
+        };
+
+    let conn = match req_state(&req).db.get_conn().map_err(Error::DbError) {
+        Ok(conn_ref) => conn_ref,
+        Err(err) => return err.into(),
+    };
+
+    let ident = PackageIdent::new(origin.clone(), pkg.clone(), None, None);
+
+    let lpr = ListPackages { ident:      BuilderPackageIdent(ident.clone()),
+                             visibility: helpers::visibility_for_optional_session(&req,
+                                                                                  Some(account_id),
+                                                                                  &ident.origin),
+                             page:       1 as i64,
+                             limit:      50 as i64, };
+
+    // Validate that no packages for the setting still exists
+    match Package::list(lpr, &*conn).map_err(Error::DieselError) {
+        Ok((_packages, count)) => {
+            if count > 0 {
+                return HttpResponse::new(StatusCode::UNPROCESSABLE_ENTITY);
```

Comment:
While technically probably an appropriate error, I personally believe that a 409 is a bit more helpful in this context. After reading the [RFC](https://tools.ietf.org/html/rfc4918#section-11.2) on 422, I understand it as: the request and the included entity was syntactically correct but semantically didn't make sense. Whereas a 409: the request could not be completed due to a conflict with the current state of the target resource.

A pattern that we used on the origin deletion was to respond with a 409 and also [embelish](https://github.com/habitat-sh/builder/blob/master/components/builder-api/src/server/resources/origins.rs#L315-L317) the response with a [body](https://github.com/habitat-sh/builder/blob/76ec0d7a3dc05a3766380a673d523c62acf1043e/test/builder-api/src/origins.js#L381) containing information for the user-agent to resolve the conflict. It would be a bit more work but may be a more helpful UX when we implement the hab CLI portion of this. Thoughts?

Reply:
Yeah I like that Idea. I'll get that into the work I'm doing now with wiring this into the deletion process. Will push that up shortly.